src/SFA.DAS.AdminService.Web/Services/ApplicationService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using SFA.DAS.AdminService.Web.Domain.Apply;
+using SFA.DAS.AdminService.Web.Infrastructure;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.ApplyTypes;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AdminService.Web.Services
+{
+    public class ApplicationService : IApplicationService
+    {
+        private readonly IApiClient _apiApiClient;
+        private readonly IApplicationApiClient _applyApiClient;
+        private readonly IQnaApiClient _qnaApiClient;
+
+        public ApplicationService(IApiClient apiClient, IApplicationApiClient applyApiClient, IQnaApiClient qnaApiClient)
+        {
+            _apiApiClient = apiClient;
+            _applyApiClient = applyApiClient;
+            _qnaApiClient = qnaApiClient;
+        }
+
+        public async Task<WithdrawalApplicationDetails> GetWithdrawalApplicationDetails(Guid applicationId)
```

Comment:
The return type should indicate that a null return is possible. The compiler then marks the calling line as "maybe null", and code analysis tools can make use of this
```suggestion
        public async Task<WithdrawalApplicationDetails?> GetWithdrawalApplicationDetails(Guid applicationId)
```

Reply:
I just noticed that a few changes to the project file would be needed to make use of nullables. If you decide to do this, you would have to add these lines to the PropertyGroup section of the csproj:

```
<LangVersion>10.0</LangVersion> <!-- 8.0 or higher will work, but .NET 6 supports C# 10-->
<Nullable>enable</Nullable>
```

This should've been done as part of the .NET 6 upgrade. It's unfortunate that our .NET 6 apps are still using the version of C# released with .NET Core 2.0!

Reply:
I'll leave these for now then